backend/ohq/permissions.py

```diff
@@ -503,3 +503,107 @@ def has_permission(self, request, view):
             return True
 
         return True
+
+class DocumentCreatePermission(permissions.BasePermission):
+    def has_permission(self, request, view):
+        if not request.user.is_authenticated:
+            return False
+
+        membership = Membership.objects.filter(
+            course=view.kwargs["course_pk"], user=request.user
+        ).first()
+
+        # Non-Students can't do anything
+        if membership is None:
+            return False
+
+        if view.action == "retrieve":
+            return membership.is_ta
```

Comment:
I think this would only mean that TAs roles can have permission to retrieve but not Leadership roles (like Head TA or Professor), so should be:

`return membership.is_ta or membership.is_leadership`